routes/aggrtimeout.go

```diff
@@ -6,34 +6,50 @@ import (
 	"strings"
 )
 
-func ConfigureAggrTimeout(route *InputRoute) *InputRoute {
+func parseTimeouts(v string) (int, error) {
 	var timeout int
 	var err error
 
 	times := map[string]int{
 		"s": 1,
 		"m": 60,
 		"h": 3600,
+		"d": 86400,
 	}
 
-	if len(route.Plugins.AggregateIssuesTimeout) > 0 {
-		wasConvert := false
-		for suffix, k := range times {
-			if strings.HasSuffix(strings.ToLower(route.Plugins.AggregateIssuesTimeout), suffix) {
-				timeout, err = strconv.Atoi(strings.TrimSuffix(route.Plugins.AggregateIssuesTimeout, suffix))
-				timeout *= k
-				wasConvert = true
-				break
-			}
+	wasConvert := false
+	for suffix, k := range times {
+		if strings.HasSuffix(strings.ToLower(v), suffix) {
+			timeout, err = strconv.Atoi(strings.TrimSuffix(v, suffix))
+			timeout *= k
+			wasConvert = true
+			break
 		}
-		if !wasConvert {
-			timeout, err = strconv.Atoi(route.Plugins.AggregateIssuesTimeout)
+	}
+	if !wasConvert {
+		timeout, err = strconv.Atoi(v)
+	}
+	return timeout, err
+}
+
+func ConfigureTimeouts(route *InputRoute) *InputRoute {
+	if len(route.Plugins.AggregateMessageTimeout) > 0 {
+		aggregateTimeoutSeconds, err := parseTimeouts(route.Plugins.AggregateMessageTimeout)
+		if err != nil {
+			log.Printf("%q settings: Can't convert 'aggregate-message-timeout'(%q) to seconds.",
+				route.Name, route.Plugins.AggregateMessageTimeout)
 		}
+
+		route.Plugins.AggregateTimeoutSeconds = aggregateTimeoutSeconds
```

Comment:
is this string really need?
```go
	var err error
	route.Plugins.UniqueMessageTimeoutSeconds, err = parseTimeouts(route.Plugins.AggregateMessageTimeout)
	if err != nil {
		log.Printf("%q settings: Can't convert 'aggregate-message-timeout'(%q) to seconds.",
			route.Name, route.Plugins.AggregateMessageTimeout)
	}
```

Reply:
I've got an compilation error with code you suggested

Reply:
it works for me. checked.
```go
		var err error
		route.Plugins.AggregateTimeoutSeconds, err = parseTimeouts(route.Plugins.AggregateMessageTimeout)
		if err != nil {
			log.Printf("%q settings: Can't convert 'aggregate-message-timeout'(%q) to seconds.",
				route.Name, route.Plugins.AggregateMessageTimeout)
		}
```

Reply:
it doesn't matter.